app/client/service_runner.py

```diff
@@ -88,7 +85,7 @@ def is_queue_empty():
             raise Exception('Master service did not become idle before timeout.')
 
     def kill(self):
-        self._run_service('{} stop'.format(self._main_executable))
+        self._run_service([self._main_executable, 'stop'])
 
     def _run_service(self, cmd, service_url=None):
```

Comment:
I don't quite understand this. Are you suggesting break the string into cmd sequence in Popen_with_delayed_expansion instead of here?